core/space/services/services_vault.go

```diff
@@ -65,10 +54,5 @@ func (s *Space) RecoverKeysByLocalBackup(ctx context.Context, path string) error
 		return err
 	}
 
-	// Restore Textile Client state
```

Comment:
Should we still overwrite the thread ID in the store here? Thinking otherwise `findOrCreateMetaThreadID` will return the existing thread ID from an old key if a recovery was done on top of it.

Reply:
Changed the imp of `findOrCreateMetaThreadID` in 4944baf to use the key pair as the store key, that way it will keep all threads id and return the correct one for the current key being used.